pkg/metrics/metricsource.go

```diff
@@ -152,6 +153,8 @@ func (ms MetricSource) String() string {
 		return "network"
 	case MetricSourceSnmp:
 		return "snmp"
+	case MetricSourceAgentTelemetry:
+		return "agent_telemetry"
```

Comment:
Why different name? Would it make sense to return the original core check name here?

Reply:
I chose this since I thought `telemetry` was a bit too ambiguous and this keeps consistency with the `MetricSource*` pattern.
Another option would be the upstream string value for this `OriginService`, and that value is `internal`.

Do you have a preference here? I don't really have any strong preference.